src/smt/theory_datatype.cpp

```diff
@@ -410,6 +411,59 @@ namespace smt {
         return r;
     }
 
+    // explain the cycle root -> … -> app -> root
+    void theory_datatype::occurs_check_explain(enode * app, enode * const root) {
```

Comment:
Do you really intend to say that enode * const root instead of enode const* root?